src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -198,7 +212,25 @@ protected Solution SetCurrentSolution(Solution solution)
         /// <param name="projectId">The id of the project updated by <paramref name="transformation"/> to be passed to the workspace change event.</param>
         /// <param name="documentId">The id of the document updated by <paramref name="transformation"/> to be passed to the workspace change event.</param>
         /// <returns>True if <see cref="CurrentSolution"/> was set to the transformed solution, false if the transformation did not change the solution.</returns>
-        internal bool SetCurrentSolution(Func<Solution, Solution> transformation, WorkspaceChangeKind kind, ProjectId? projectId = null, DocumentId? documentId = null)
+        internal bool SetCurrentSolution(
+            Func<Solution, Solution> transformation,
+            WorkspaceChangeKind kind,
+            ProjectId? projectId = null,
+            DocumentId? documentId = null)
+        {
+            return SetCurrentSolution(transformation, static _ => { }, kind, projectId, documentId);
+        }
+
+        /// <inheritdoc cref="SetCurrentSolution(Func{Solution, Solution}, WorkspaceChangeKind, ProjectId?, DocumentId?)"/>
+        /// <param name="onAfterUpdate">Callback to run only once the solution returned by <paramref
+        /// name="transformation"/> has been assigned as the <see cref="CurrentSolution"/> of this <see
+        /// cref="Workspace"/>.</param>
+        internal bool SetCurrentSolution(
+            Func<Solution, Solution> transformation,
+            Action<Solution> onAfterUpdate,
+            WorkspaceChangeKind kind,
+            ProjectId? projectId = null,
+            DocumentId? documentId = null)
```

Comment:
added this helper as there are a couple of callers that wanted to update the solution, then do something else when that succeeds, then report events.  This enables that pattern.